newIDE/app/src/EventsSheet/InstructionEditor/NewInstructionEditorDialog.js

```diff
@@ -146,6 +171,30 @@ export default function NewInstructionEditorDialog({
     }
   };
 
+  const _addBehavior = (type: string, defaultName: string) => {
+    if (!chosenObject) return;
+    setNewBehaviorDialogOpen(false);
+
+    addBehaviorToObject(project, chosenObject, type, defaultName);
+
+    // Re-choose the same object to force recomputation of chosenObjectInstructionsInfoTree
+    chooseObject(chosenObject.getName());
```

Comment:
is this a hack? I'm not familiar with this panel, just wondering if there's a function for that, like 'forceUpdate'

Reply:
Not really a hack, It's just that the computation of displayed instructions only happens when you choose an object, not when a change in the object behaviors has been detected. And this change happens at the C++ level so there is not much we can do to make this refresh cleaner.

Reply:
Let's add a comment "This is not done because a change in the object behaviors is not detected by React at this level" or something like this :)

Reply:
(One could argue we could pass a key={/*object name + all the names and type of the behaviors in an attempt to create a key reflecting the behaviors of the object*/} - but unsure if this is more clean/efficient or not).